ponzu/llama/curatedTVL.py

```diff
@@ -0,0 +1,130 @@
+        df = df[~df['category'].isin(exclude_categories)]
+        exclude_protocols = ['ribbon','frax','unslashed','sturdy','alongside','illuminate','instadapp','defisaver','pooltogether','cian','ubiquity-dao','sandclock','templedao','unbound','axia-protocol','tornado-cash','aztec','tronnrg','garble.money','sperax-demeter','volta-finance','uniwswap-unia-farms','mm-stableswap-polygon','zkbob','brokkr-finance','defrost','sherpa-cash','sturdy','shadecash','zeta']
+        df = df[~df['protocol'].isin(exclude_protocols)]
+        self.curated_tvl = df
+        return df
+    
+    def _prepare_tvl(self, chains, curated):
+        if curated:
+            self.tvl = self.curate_tvl(chains)
+        else:
+            self.tvl = self.fetch_tvl(chains)
+
+    def _get_pivot_table(self, values, index, columns, aggfunc, tvl=None):
+        if tvl is None:
+            tvl = self.tvl
+        return pd.pivot_table(tvl, values=values, index=index, columns=columns, aggfunc=aggfunc)
+
+    def chain_tvl(self, chains, curated=False):
+        self._prepare_tvl(chains, curated)
+        return self._get_pivot_table('tvl', 'date', 'chain', sum)
+
+    def chain_protocols_tvl(self, chains, curated=False):
+        self._prepare_tvl(chains, curated)
+        tvl_protocols = {}
+        for chain in chains:
+            tvl_protocols_chain = self.tvl[self.tvl['chain'] == chain]
+            tvl_protocols[f'tvl_protocols_{chain}'] = pd.pivot_table(tvl_protocols_chain, values='tvl', index='date', columns='protocol', aggfunc=sum)
+
+        return tvl_protocols
+
+    def chain_tvb(self, chains, curated=False):
+        self._prepare_tvl(chains, curated)
+        df_borrows = self.tvl[self.tvl['type'] == 'borrowed']
+        return self._get_pivot_table('tvl', 'date', 'chain', sum, df_borrows)
+
+    def chain_protocols_tvb(self, chains, curated=False):
+        self._prepare_tvl(chains, curated)
+        tvb_protocols = {}
+        for chain in chains:
+            tvb_protocols_chain = self.tvl.loc[(self.tvl['type'] == 'borrowed') & (self.tvl['chain'] == chain)]
+            tvb_protocols[f'tvb_protocols_{chain}'] = pd.pivot_table(tvb_protocols_chain, values='tvl', index='date', columns='protocol', aggfunc=sum)
+        return tvb_protocols
+
+    def chain_liquid_staking(self, chains):
+        self._prepare_tvl(chains, False) 
+        df_liquid_staking = self.tvl[self.tvl['category'] == 'Liquid Staking']
+        return self._get_pivot_table('tvl', 'date', 'chain', sum, df_liquid_staking)
+    
+    def chain_protocols_liquid_staking(self, chains):
+        self._prepare_tvl(chains, False)
+        df_liquid_staking = self.tvl[self.tvl['category'] == 'Liquid Staking']
+        ls_protocols = {}
+        for chain in chains:
+            ls_chain = df_liquid_staking[df_liquid_staking['chain'] == chain]
+            ls_protocols[f'ls_protocols_{chain}'] = pd.pivot_table(ls_chain, values='tvl', index='date', columns='protocol', aggfunc=sum)
```

Comment:
why are you adding 'ls_protocols_' to the dict key - woulddnt anyone calling the liquid staking fn know this is liquid staking and adding the bonus info to the key make it harder to consume after you call?

Reply:
I guess so, can remove